src/rules/dollar-variable-colon-space-before/index.js

```diff
@@ -0,0 +1,34 @@
+import {
+  namespace,
+  whitespaceChecker,
+} from "../../utils"
+import { utils } from "stylelint"
+import { variableColonSpaceChecker } from "../dollar-variable-colon-space-after"
```

Comment:
Oops, sorry, merged before this comment appeared on the page.

As for the question - the rules' logic is ought to be very similar (actually, almost the same except for where the space is checked and the messages). It is quite a normal practice in stylelint (like [here](https://github.com/stylelint/stylelint/blob/master/src/rules/declaration-colon-space-before/index.js#L15)), and our rules use it too (like the neighbouring sass operators rules).